cloudoperators/juno
# fix(heureka): corrects URL state handling for filter property

# Summary

Fixed an issue where activeFilters were not immediately applied after URL updates, requiring additional actions to trigger synchronization. Refactored useUrlState by updating useEffect dependencies to ensure real-time syncing of URL state and filters when changes occur.
Additionally, restored the missing activeFilters prop in FilterSelect within the Filters control.

Furthermore, the detailsFor prop is now handled correctly, ensuring that when a detail panel is closed, the corresponding detailsFor information is removed from the URL, maintaining a clean and accurate URL state.

# Changes Made

- Updated useEffect dependencies in useUrlState for immediate filter synchronization.
- Restored the activeFilters prop in FilterSelect within the Filters control.
- Added logic to remove detailsFor and associated details from the URL when the panel is closed.

# Related Issues

- https://github.com/cloudoperators/heureka/issues/375


# Testing Instructions

- Apply filters on different views in the app.
- Verify that the selected filters for each view are reflected in the URL.

For example, the following part of URL should display the `Support group ccrn = compute-storage-api` filter on the `Services` view:
f:(Components:(:),IssueMatches:(:),Services:(supportGroupCcrn:(compute~Fstorage~Fapi)))

1. `npm i`
2. `npm run TASK`

# Checklist
